src/nvim/window.c

```diff
@@ -1741,6 +1741,11 @@ void close_windows(buf_T *buf, int keep_curwin)
     } else
       wp = wp->w_next;
   }
+  if (count != tabpage_index(NULL)) {
+    char_u prev_idx[NUMBUFLEN];
+    sprintf((char *)prev_idx, "%i", tabpage_index(curtab));
```

Comment:
Should use snprintf. Also might as well check `has_event(EVENT_TABCLOSED)`, no?

Reply:
This was copied/moved IIRC.
Changed it now, with tests added.